lib/pure/strutils.nim

```diff
@@ -1438,43 +1438,49 @@ func unindent*(s: string, count: Natural = int.high,
     result.add(line[indentCount*padding.len .. ^1])
     i.inc
 
-func indentation*(s: string): Natural {.since: (1, 3).} =
-  ## Returns the amount of indentation all lines of `s` have in common,
-  ## ignoring lines that consist only of whitespace.
-  result = int.high
-  for line in s.splitLines:
-    for i, c in line:
-      if i >= result: break
-      elif c != ' ':
-        result = i
-        break
-  if result == int.high:
-    result = 0
-
-func dedent*(s: string, count: Natural = indentation(s)): string {.rtl,
-    extern: "nsuDedent", since: (1, 3).} =
-  ## Unindents each line in `s` by `count` amount of `padding`.
-  ## The only difference between this and the
-  ## `unindent func<#unindent,string,Natural,string>`_ is that this by default
-  ## only cuts off the amount of indentation that all lines of `s` share as
-  ## opposed to all indentation. It only supports spaces as padding.
-  ##
-  ## **Note:** This does not preserve the new line characters used in `s`.
-  ##
-  ## See also:
-  ## * `unindent func<#unindent,string,Natural,string>`_
-  ## * `align func<#align,string,Natural,char>`_
-  ## * `alignLeft func<#alignLeft,string,Natural,char>`_
-  ## * `spaces func<#spaces,Natural>`_
-  ## * `indent func<#indent,string,Natural,string>`_
-  runnableExamples:
-    let x = """
-      Hello
-        There
-    """.dedent()
-
-    doAssert x == "Hello\n  There\n"
-  unindent(s, count, " ")
+since (1, 3):
+  func indentation*(s: string): Natural =
+    ## Returns the amount of indentation all lines of `s` have in common,
+    ## ignoring lines that consist only of whitespace.
+    result = int.high
+    for line in s.splitLines:
+      for i, c in line:
+        if i >= result: break
+        elif c != ' ':
+          result = i
+          break
+    if result == int.high:
+      result = 0
+
+  func dedent*(s: string, count: Natural): string {.rtl, extern: "nsuDedent".} =
+    unindent(s, count, " ")
+
+  proc dedent*(a: string): string =
+    ## Unindents each line in `s` by `count` amount of `padding`.
+    ## The only difference between this and the
+    ## `unindent func<#unindent,string,Natural,string>`_ is that this by default
+    ## only cuts off the amount of indentation that all lines of `s` share as
+    ## opposed to all indentation. It only supports spaces as padding.
+    ##
+    ## **Note:** This does not preserve the new line characters used in `s`.
+    ##
+    ## See also:
+    ## * `unindent func<#unindent,string,Natural,string>`_
+    ## * `align func<#align,string,Natural,char>`_
+    ## * `alignLeft func<#alignLeft,string,Natural,char>`_
+    ## * `spaces func<#spaces,Natural>`_
+    ## * `indent func<#indent,string,Natural,string>`_
+    # workaround, see https://github.com/timotheecour/Nim/issues/521
+    runnableExamples:
+      let x = """
+        Hello
+          There
+      """.dedent()
+
+      doAssert x == "Hello\n  There\n"
+    let b = a
+    let i = indentation(b)
+    dedent(a, i)
```

Comment:
why not simply:
```suggestion
    dedent(a, indentation(a))
```
?
(it's not a template, no risk of multiple evaluation bug)

also, `proc dedent*(a: string): string =` should be {.inline.}

Reply:
Thank you, I added this change. (from my name — sorry, I did not notice button "commit suggestion". I'll be more mindful in the future)

Reply:
>  I'll be more mindful in the future

not a concern, the commits get squashed when the PR is merged anyways; I also do the same sometimes so I can keep rebasing instead of merging